### PR TITLE
Fix Documentation for Customizing Metrics and  TCP Metrics Task

### PR DESCRIPTION
**Pull request by gargnupur**
Fix Documentation for Customizing Metrics and  TCP Metrics Task
Ref: https://github.com/istio/istio/issues/23782
Ref: https://github.com/istio/istio.io/issues/7184
Ref: https://github.com/istio/istio.io/issues/7280

### CODE REVIEW

File: content/en/docs/tasks/observability/metrics/customize-metrics/index.md [hunk 1]
```diff
@@ -86,7 +86,7 @@ dimensions.
             {
                 "name": "requests_total",
                 "dimensions": {
-                    "destination_port": "destination.port",
+                    "destination_port": "string(destination.port)",
```
Comment (shamsher31) [inline content/en/docs/tasks/observability/metrics/customize-metrics/index.md:89]: Not sure but is this valid conversion in `JSON` ?

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md [hunk 2]
```diff
@@ -91,15 +91,17 @@ example configuration and commands.
...
-
     For the Bookinfo sample, visit `http://$GATEWAY_URL/productpage` in your web
     browser or use the following command:
 
     {{< text bash >}}
     $ curl http://$GATEWAY_URL/productpage
     {{< /text >}}
 
+    {{< tip >}}
+    `$GATEWAY_URL` is the value set in the [Bookinfo](/docs/examples/bookinfo/) example.
```
Comment (shamsher31) [inline content/en/docs/tasks/observability/metrics/tcp-metrics/index.md:102]: I think pointing users to the right section would be more helpful.

So this 
`/docs/examples/bookinfo/`

Should be this
`docs/setup/getting-started/#determining-the-ingress-ip-and-ports`

File: content/en/docs/tasks/observability/metrics/customize-metrics/index.md (hunk 1 above)
Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/customize-metrics/index.md:89]: the conversion will happen on the proxy side.. this is an expression that istio/proxy will handle...

File: content/en/docs/tasks/observability/metrics/tcp-metrics/index.md (hunk 2 above)
Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/tcp-metrics/index.md:102]: We had this before but it was confusing to folks.. see removed changes please... 
Ref: https://github.com/istio/istio.io/issues/7184

File: content/en/docs/tasks/observability/metrics/customize-metrics/index.md [hunk 3]
```diff
@@ -100,21 +100,6 @@ dimensions.
...
-   command:
-
-    {{< text yaml >}}
-    apiVersion: extensions/v1beta1
-    kind: Deployment
-    spec:
-    template:
-        metadata:
-        annotations:
-            sidecar.istio.io/extraStatTags: destination_port,request_host
```
Comment (mandarjog) [inline content/en/docs/tasks/observability/metrics/customize-metrics/index.md:115]: is this not needed anymore?
apart from this lgtm.

Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/customize-metrics/index.md:115]: we see the tags in prometheus without this and with this, there is a crash... thus, removing it..

Comment (douglas-reid) [reply content/en/docs/tasks/observability/metrics/customize-metrics/index.md:115]: this is ok for `destination_port` and a few others, but isn't this required for anything non-standard ?  how do we want to handle that?

Comment (frankbu) [reply content/en/docs/tasks/observability/metrics/customize-metrics/index.md:115]: @douglas-reid I'm not sure I understand if this comment affects the merge-ability of this PR or not? Are you suggesting a change might be needed in this PR, or is this one good to go as is?

Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/customize-metrics/index.md:115]: @douglas-reid : do we know the attributes for whom this is not needed? 
or I can add it in a note that. if you are using it for anything non-standard, then you might want to add this?

Comment (douglas-reid) [reply content/en/docs/tasks/observability/metrics/customize-metrics/index.md:115]: i think this is needed for anything not in this list: https://github.com/istio/istio/blob/c24438d1706e4db7ffdacd2f686223f7a6c104c0/pkg/bootstrap/config.go#L188

Comment (douglas-reid) [reply content/en/docs/tasks/observability/metrics/customize-metrics/index.md:115]: NOTE: I have not verified.

Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/customize-metrics/index.md:115]: Thanks Doug.. this looks right.. let me add this link and not delete the extraStatTags para...

Comment (gargnupur) [reply content/en/docs/tasks/observability/metrics/customize-metrics/index.md]: Fixed..PTAL

Comment (frankbu) [reply content/en/docs/tasks/observability/metrics/customize-metrics/index.md]: Seems strange that the tip says it's only needed for things not it DefaultStatTags, but then the example shows two, both of which are in DefaultStatTags.

Anyway, @douglas-reid, do you think we should merge this now?

Comment (douglas-reid) [reply content/en/docs/tasks/observability/metrics/customize-metrics/index.md]: Yes. Let's merge this now. We can improve the docs incrementally.

**Pull request merged**